src/index/mod.rs

```diff
@@ -198,16 +198,32 @@ impl IndexSearcher {
         config: &SearchConfig,
     ) -> Result<Box<dyn tv::query::Query>, tv::TantivyError> {
         let mut keys = Vec::new();
+        let mut parts = Vec::new();
 
-        let term = if let Some(room) = &config.room_id {
+        if let Some(rooms) = &config.room_ids {
+            let mut room_parts = Vec::new();
             keys.push(self.room_id_field);
-            format!("+room_id:\"{}\" AND \"{}\"", room, term)
-        } else if term.is_empty() {
-            "*".to_owned()
-        } else {
-            term.to_owned()
+            for room in rooms {
+                room_parts.push(format!("room_id:\"{}\"", room))
```

Comment:
This seems sensible, but did you test that this works as expected? I'm not sure how the query parser reacts to this. As I mentioned, having tests for this would be very welcome.

Reply:
I black-box-tested it with radical-native (and now added a test)